libcudacxx/test/libcudacxx/cuda/ptx/manual/shfl_test.h

```diff
@@ -0,0 +1,139 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of libcu++, the C++ Standard Library for your entire system,
+// under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2025 NVIDIA CORPORATION & AFFILIATES.
+//
+//===----------------------------------------------------------------------===//
+// UNSUPPORTED: libcpp-has-no-threads
+// UNSUPPORTED: clang && !nvcc
+
+// <cuda/ptx>
+
+__host__ __device__ void test_shfl_full_mask()
+{
+#if __cccl_ptx_isa >= 600 && __CUDA_ARCH__
+  constexpr unsigned FullMask = 0xFFFFFFFF;
+  auto data                   = threadIdx.x;
+  bool pred1, pred2, pred3, pred4;
+  auto res1 = cuda::ptx::shfl_sync_idx(data, pred1, 2 /*idx*/, 0b11111 /*clamp*/, FullMask);
+  _CCCL_ASSERT(res1 == 2 && pred1, "shfl_mode_idx failed");
```

Comment:
I think we use plain `assert` in libcu++ tests. Why is `_CCCL_ASSERT` needed here, since it is affected by different cmake build options.